component/status.go

```diff
@@ -0,0 +1,106 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component
+
+import (
+	"errors"
+	"time"
+)
+
+type Status int
```

Comment:
nit: use int32 to guarantee size independent of the platform.

Reply:
Why do we need to guarantee the size?

Reply:
Is not about guaranteeing the size, is about best practice, we definitely don't need this to be int64 on a 64bit arch. And in general we use int32 for enums, gives enough room to grow and saves memory.

Reply:
I know that it does not happen in this case of alignment situation, but in lots of cases this can be paired with another enum/int32 so may save memory there.

I think in vast majority of the cases struct members should be order by size and in that case the alignment rules should allow us to save that memory.

Reply:
It is a best practice in general "." Not wast memory, and indeed you don't always save, but with `int` you always waste :)

Reply:
Changed to int32.